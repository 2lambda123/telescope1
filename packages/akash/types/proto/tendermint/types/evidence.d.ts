import Long from "long";
import * as _m0 from "protobufjs/minimal";
import { Vote, LightBlock } from "../../tendermint/types/types";
import { Validator } from "../../tendermint/types/validator";
export interface Evidence {
    duplicateVoteEvidence: DuplicateVoteEvidence | undefined;
    lightClientAttackEvidence: LightClientAttackEvidence | undefined;
}
/** DuplicateVoteEvidence contains evidence of a validator signed two conflicting votes. */
export interface DuplicateVoteEvidence {
    voteA: Vote;
    voteB: Vote;
    totalVotingPower: Long;
    validatorPower: Long;
    timestamp: Date;
}
/** LightClientAttackEvidence contains evidence of a set of validators attempting to mislead a light client. */
export interface LightClientAttackEvidence {
    conflictingBlock: LightBlock;
    commonHeight: Long;
    byzantineValidators: Validator[];
    totalVotingPower: Long;
    timestamp: Date;
}
export interface EvidenceList {
    evidence: Evidence[];
}
export declare const Evidence: {
    encode(message: Evidence, writer?: _m0.Writer): _m0.Writer;
    decode(input: _m0.Reader | Uint8Array, length?: number): Evidence;
    fromJSON(object: any): Evidence;
    toJSON(message: Evidence): unknown;
    fromPartial<I extends unknown>(object: I): Evidence;
};
export declare const DuplicateVoteEvidence: {
    encode(message: DuplicateVoteEvidence, writer?: _m0.Writer): _m0.Writer;
    decode(input: _m0.Reader | Uint8Array, length?: number): DuplicateVoteEvidence;
    fromJSON(object: any): DuplicateVoteEvidence;
    toJSON(message: DuplicateVoteEvidence): unknown;
    fromPartial<I extends unknown>(object: I): DuplicateVoteEvidence;
};
export declare const LightClientAttackEvidence: {
    encode(message: LightClientAttackEvidence, writer?: _m0.Writer): _m0.Writer;
    decode(input: _m0.Reader | Uint8Array, length?: number): LightClientAttackEvidence;
    fromJSON(object: any): LightClientAttackEvidence;
    toJSON(message: LightClientAttackEvidence): unknown;
    fromPartial<I extends unknown>(object: I): LightClientAttackEvidence;
};
export declare const EvidenceList: {
    encode(message: EvidenceList, writer?: _m0.Writer): _m0.Writer;
    decode(input: _m0.Reader | Uint8Array, length?: number): EvidenceList;
    fromJSON(object: any): EvidenceList;
    toJSON(message: EvidenceList): unknown;
    fromPartial<I extends unknown>(object: I): EvidenceList;
};
declare type Builtin = Date | Function | Uint8Array | string | number | boolean | undefined;
export declare type DeepPartial<T> = T extends Builtin ? T : T extends Long ? string | number | Long : T extends Array<infer U> ? Array<DeepPartial<U>> : T extends ReadonlyArray<infer U> ? ReadonlyArray<DeepPartial<U>> : T extends {} ? {
    [K in keyof T]?: DeepPartial<T[K]>;
} : Partial<T>;
declare type KeysOfUnion<T> = T extends T ? keyof T : never;
export declare type Exact<P, I extends P> = P extends Builtin ? P : P & {
    [K in keyof P]: Exact<P[K], I[K]>;
} & Record<Exclude<keyof I, KeysOfUnion<P>>, never>;
export {};
