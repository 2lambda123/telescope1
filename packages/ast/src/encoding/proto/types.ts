import * as t from '@babel/types';
import { ProtoType, ProtoField } from '@osmonauts/types';
import { pascal } from 'case';
import { GenericParseContext, ProtoParseContext } from '../context';

export const SCALAR_TYPES = [
    'string',
    'double',
    'float',
    'int32',
    'uint32',
    'sint32',
    'fixed32',
    'sfixed32',
    'int64',
    'uint64',
    'sint64',
    'fixed64',
    'sfixed64',
    'bytes',
    'bool'
];

export const NATIVE_TYPES = [
    ...SCALAR_TYPES,

    // yes, they're not "native", but in many ways, they are...
    'google.protobuf.Timestamp',
    'google.protobuf.Duration',
    'google.protobuf.Any'
]

export const getTSTypeFromProtoType = (context: GenericParseContext, type: string) => {
    switch (type) {
        case 'string':
            return t.tsStringKeyword();
        case 'double':
        case 'float':
        case 'int32':
        case 'uint32':
        case 'sint32':
        case 'fixed32':
        case 'sfixed32':
            return t.tsNumberKeyword();
        case 'int64':
        case 'uint64':
        case 'sint64':
        case 'fixed64':
        case 'sfixed64':
            return t.tsTypeReference(t.identifier('Long'))
        case 'bytes':
            return t.tsTypeReference(t.identifier('Uint8Array'));
        case 'bool':
            return t.tsBooleanKeyword();
        case 'google.protobuf.Timestamp':
            switch (context.options.useDate) {
                case 'timestamp':
                    return t.tsTypeReference(t.identifier('Timestamp'));
                case 'date':
                default:
                    return t.tsTypeReference(t.identifier('Date'));
            }
        case 'google.protobuf.Duration':
            switch (context.options.useDuration) {
                case 'duration':
                    return t.tsTypeReference(t.identifier('Duration'));
                case 'string':
                default:
                    return t.tsStringKeyword();
            }
        case 'google.protobuf.Any':
            return t.tsTypeReference(t.identifier('Any'));
        default:
            throw new Error('getTSTypeFromProtoType() type not found');
    };
};

// https://github.com/protobufjs/protobuf.js/blob/master/src/types.js#L38-L54
export const types = {
    basic: {
        double: 1,
        float: 5,
        int32: 0,
        uint32: 0,
        sint32: 0,
        fixed32: 5,
        sfixed32: 5,
        int64: 0,
        uint64: 0,
        sint64: 0,
        fixed64: 1,
        sfixed64: 1,
        bool: 0,
        string: 2,
        bytes: 2
    },
    defaults: {
        double: 0,
        float: 0,
        int32: 0,
        uint32: 0,
        sint32: 0,
        fixed32: 0,
        sfixed32: 0,
        int64: 0,
        uint64: 0,
        sint64: 0,
        fixed64: 0,
        sfixed64: 0,
        bool: false,
        string: '',
        bytes: [],
        undefined: null
    },
    long: { int64: 0, uint64: 0, sint64: 0, fixed64: 1, sfixed64: 1 },
    mapKey: {
        int32: 0,
        uint32: 0,
        sint32: 0,
        fixed32: 5,
        sfixed32: 5,
        int64: 0,
        uint64: 0,
        sint64: 0,
        fixed64: 1,
        sfixed64: 1,
        bool: 0,
        string: 2
    },
    packed: {
        double: 1,
        float: 5,
        int32: 0,
        uint32: 0,
        sint32: 0,
        fixed32: 5,
        sfixed32: 5,
        int64: 0,
        uint64: 0,
        sint64: 0,
        fixed64: 1,
        sfixed64: 1,
        bool: 0
    }
};
export const getWireNumber = (type) => {
    if (types.basic.hasOwnProperty(type)) {
        return types.basic[type];
    }
    return 2;
};

export const getPackedWireNumber = (type) => {
    if (types.packed.hasOwnProperty(type)) {
        return types.packed[type];
    }
    return 2;
};

export const getTagNumber = (field: ProtoField) => {
    let wire = getWireNumber(field.type);
    if (field.parsedType.type === 'Enum') {
        wire = 0;
    }
    if (field.rule === 'repeated') {
        wire = 2;
    }
    return ((field.id << 3) | wire) >>> 0;
};


export const getDefaultTSTypeFromProtoType = (
    context: ProtoParseContext, // here for future forceLong=string
    field: ProtoField,
    isOptional: boolean
) => {

    if (isOptional) {
        return t.identifier('undefined');
    }

    if (field.rule === 'repeated') {
        return t.arrayExpression([]);
    }

    if (field.keyType) {
        return t.objectExpression([])
    }

    if (field.parsedType?.type === 'Enum') {
        return t.numericLiteral(0);
    }

    switch (field.type) {
        case 'string':
            return t.stringLiteral('');
        case 'double':
        case 'float':
        case 'int32':
        case 'uint32':
        case 'sint32':
        case 'fixed32':
        case 'sfixed32':
            return t.numericLiteral(0);
        case 'uint64':
            return t.memberExpression(
                t.identifier('Long'),
                t.identifier('UZERO')
            );
        case 'int64':
        case 'sint64':
        case 'fixed64':
        case 'sfixed64':
            return t.memberExpression(
                t.identifier('Long'),
                t.identifier('ZERO')
            );
        case 'bytes':
            return t.newExpression(
                t.identifier('Uint8Array'),
                []
            );
        case 'bool':
            return t.booleanLiteral(false);

        // OTHER TYPES
        case 'google.protobuf.Timestamp':
            return t.identifier('undefined');
        case 'google.protobuf.Duration':
            return t.identifier('undefined');
        case 'google.protobuf.Any':
            return t.identifier('undefined');

        case 'cosmos.base.v1beta1.Coins':
            return t.arrayExpression([]);
        case 'cosmos.base.v1beta1.Coin':
            return t.identifier('undefined');

        default:
            // console.warn('getDefaultTSTypeFromProtoType() type not found: ' + type);
            return t.identifier('undefined');
    };
};

const lowerFirst = (str: string) => {
    return str.charAt(0).toLowerCase() + str.substring(1);
};
const upperFirst = (str: string) => {
    return str.charAt(0).toUpperCase() + str.substring(1);
};

export const getEnumToJsonName = (name) => {
    return lowerFirst(name) + 'ToJSON';
};

export const getEnumFromJsonName = (name) => {
    return lowerFirst(name) + 'FromJSON';
};

export const getFieldsTypeName = (field: ProtoField) => {
    if (field?.scope.length <= 1) return field.parsedType.name;
    const [_pkg, ...scopes] = field.scope;
    return [...scopes, field.parsedType.name].join('_');
};

export const getKeyTypeEntryName = (typeName: string, prop: string) => {
    return `${typeName}_${pascal(prop)}Entry`;
};

export const getBaseCreateTypeFuncName = (name) => {
    return `createBase${upperFirst(name)}`;
};

export const getOneOfs = (type: ProtoType) => {
    const keys = Object.keys(type.oneofs ?? {});
    if (!keys.length) return [];
    if (keys.length !== 1) throw new Error('getOneOfs() oneofs cardinality not known!');
    return type.oneofs[keys[0]].oneof;
};

export const getFieldOptionality = (field: ProtoField, isOneOf: boolean) => {
    return isOneOf || field?.options?.['(gogoproto.nullable)'];
};

export const getObjectNameOld = (name: string, scope: string[] = []) => {
    if (!scope.length || scope.length === 1) return name;
    const [_pkg, ...scopes] = scope;
    return [...scopes, name].join('_')
};
