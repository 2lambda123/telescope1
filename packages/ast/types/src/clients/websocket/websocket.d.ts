import * as t from '@babel/types';
import { GenericParseContext } from '../../encoding';
export declare const createWebSocketClient: (context: GenericParseContext) => t.ExportNamedDeclaration;
