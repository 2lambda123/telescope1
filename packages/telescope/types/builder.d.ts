import { ProtoStore } from '@osmonauts/proto-parser';
import { TelescopeParseContext } from './build';
import { TelescopeOptions } from '@osmonauts/types';
import { BundlerFile, TelescopeInput } from './types';
export declare class TelescopeBuilder {
    store: ProtoStore;
    protoDirs: string[];
    outPath: string;
    options: TelescopeOptions;
    contexts: TelescopeParseContext[];
    files: string[];
    readonly converters: BundlerFile[];
    readonly lcdClients: BundlerFile[];
    readonly registries: BundlerFile[];
    constructor({ protoDirs, outPath, store, options }: TelescopeInput & {
        store?: ProtoStore;
    });
    context(ref: any): TelescopeParseContext;
    addLCDClients(files: BundlerFile[]): void;
    addRegistries(files: BundlerFile[]): void;
    addConverters(files: BundlerFile[]): void;
    build(): void;
}
