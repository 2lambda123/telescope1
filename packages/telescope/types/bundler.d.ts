import * as t from '@babel/types';
import { TelescopeParseContext } from './build';
import { TelescopeBuilder } from './builder';
import { ProtoRef } from '@osmonauts/types';
import { Bundle, BundlerFile } from './types';
export declare class Bundler {
    builder: TelescopeBuilder;
    contexts: TelescopeParseContext[];
    bundle: Bundle;
    files: string[];
    readonly converters: BundlerFile[];
    readonly lcdClients: BundlerFile[];
    readonly registries: BundlerFile[];
    constructor(builder: TelescopeBuilder, bundle: Bundle);
    addLCDClients(files: BundlerFile[]): void;
    addRegistries(files: BundlerFile[]): void;
    addConverters(files: BundlerFile[]): void;
    getFreshContext(context: TelescopeParseContext): TelescopeParseContext;
    getLocalFilename(ref: ProtoRef, suffix?: string): string;
    getFilename(localname: string): string;
    writeAst(program: t.Statement[], filename: string): void;
    addToBundle(context: TelescopeParseContext, localname: string): void;
}
