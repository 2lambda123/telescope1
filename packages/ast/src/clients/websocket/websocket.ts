import * as t from '@babel/types';
import { GenericParseContext } from '../../encoding';
import { arrowFunctionExpression, identifier, memberExpressionOrIdentifier, objectPattern } from '../../utils';

export const createWebSocketClient = (context: GenericParseContext) => {

    context.addUtil('WebSocket');
  
    return t.exportNamedDeclaration(
        t.variableDeclaration(
            "const",
            [
                t.variableDeclarator(
                    t.identifier("createWebSocketClient"),
                    t.arrowFunctionExpression(
                        [
                            identifier('wsUrl', 
                                t.tsTypeAnnotation(
                                    t.tsStringKeyword()
                                ))
                        ],
                        t.blockStatement(
                            [
                                t.returnStatement(
                                    t.newExpression(
                                        t.identifier('WebSocket'),
                                        [
                                            t.identifier('wsUrl')
                                        ]
                                    )
                                )
                            ]
                        )
                    )
                )
            ]
        )
    );
  };
  