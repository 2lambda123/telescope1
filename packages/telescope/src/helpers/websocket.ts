export const webSocket = `

import WebSocket from 'isomorphic-ws';

const send = (ws: WebSocket, method: string, params: string[]) => {
    ws.send(JSON.stringify({
      "method": method,
      "params": params,
      "id": "1",
      "jsonrpc": "2.0"
    }));
}

const onOpen = (ws: WebSocket) => {
    ws.on('open', function open() {
        let timeout = setTimeout(function () {
            console.log('Connection timed out! Please check your endpoint');
        }, 5000);
  
        // if the connection is not timed out
        if (!timeout) {
            console.log('Connected to WebSocket');
        }
    });
}

const onClose = (ws: WebSocket) => {
    ws.on('close', function close() {
        console.log('disconnected');
    });
}

const onMessage = (ws: WebSocket) => {
    let finalData = JSON.parse(data.toString('utf-8'));
        if (finalData.result.data)
            console.log("------------------------------------------------------");
            console.log(finalData.result);
}

`