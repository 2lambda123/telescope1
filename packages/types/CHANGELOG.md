# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.5.4](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.5.3...@osmonauts/types@0.5.4) (2022-07-25)

**Note:** Version bump only for package @osmonauts/types





## [0.5.3](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.5.2...@osmonauts/types@0.5.3) (2022-07-24)

**Note:** Version bump only for package @osmonauts/types





## [0.5.2](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.5.1...@osmonauts/types@0.5.2) (2022-07-23)

**Note:** Version bump only for package @osmonauts/types





## [0.5.1](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.5.0...@osmonauts/types@0.5.1) (2022-07-23)

**Note:** Version bump only for package @osmonauts/types





# [0.5.0](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.4.1...@osmonauts/types@0.5.0) (2022-07-21)

**Note:** Version bump only for package @osmonauts/types





## [0.4.1](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.4.0...@osmonauts/types@0.4.1) (2022-07-18)

**Note:** Version bump only for package @osmonauts/types





# [0.4.0](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.3.12...@osmonauts/types@0.4.0) (2022-07-15)

**Note:** Version bump only for package @osmonauts/types





## [0.3.12](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.3.11...@osmonauts/types@0.3.12) (2022-07-15)

**Note:** Version bump only for package @osmonauts/types





## [0.3.11](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.3.10...@osmonauts/types@0.3.11) (2022-07-15)

**Note:** Version bump only for package @osmonauts/types





## [0.3.10](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.3.9...@osmonauts/types@0.3.10) (2022-06-24)

**Note:** Version bump only for package @osmonauts/types





## [0.3.9](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.3.8...@osmonauts/types@0.3.9) (2022-06-24)

**Note:** Version bump only for package @osmonauts/types





## [0.3.8](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.3.7...@osmonauts/types@0.3.8) (2022-06-24)

**Note:** Version bump only for package @osmonauts/types





## [0.3.7](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.3.6...@osmonauts/types@0.3.7) (2022-06-15)

**Note:** Version bump only for package @osmonauts/types





## [0.3.6](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.3.5...@osmonauts/types@0.3.6) (2022-05-23)

**Note:** Version bump only for package @osmonauts/types





## [0.3.5](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.3.4...@osmonauts/types@0.3.5) (2022-05-23)

**Note:** Version bump only for package @osmonauts/types





## [0.3.4](https://github.com/osmosis-labs/telescope/compare/@osmonauts/types@0.3.3...@osmonauts/types@0.3.4) (2022-05-20)

**Note:** Version bump only for package @osmonauts/types





## 0.3.3 (2022-05-20)

**Note:** Version bump only for package @osmonauts/types
