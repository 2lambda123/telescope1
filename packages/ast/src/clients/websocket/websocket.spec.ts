import { ProtoStore, traverse, getNestedProto } from '@osmonauts/proto-parser'
import { defaultTelescopeOptions, ProtoService } from '@osmonauts/types';
import { getGenericParseContext, expectCode, getTestProtoStore, printCode } from '../../../test-utils/';
import { createWebSocketClient } from './websocket'

it('GRPC-Gateway Msg Client', () => {
    const context = getGenericParseContext();
    expectCode(createWebSocketClient(context));
});