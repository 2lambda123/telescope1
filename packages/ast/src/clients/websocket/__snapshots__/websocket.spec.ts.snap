// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GRPC-Gateway Msg Client 1`] = `
"export const createWebSocketClient = (wsUrl: string) => {
  return new WebSocket(wsUrl);
};"
`;
